### ISSUE TITLE
API changing

### ISSUE THREAD

**Issue by JDong820** [NosotrosNueces/mcc#84]
Do you know why using MCC sucks? Because you have to understand the underlying protocol to use handlers. Needs to be re-architect'ed, or we need another wrapper on top.

**Comment by SrsBusiness** [comment]
Right now our handlers like this

'''c
void position_handler(bot_t *bot, void *autism) {
    my_struct *p = (my_struct *)autism;
    ....
}
'''

what we need is this:
'''c
void position_handler(bot_t *bot, int x, int y, int z) {
     ....
}
'''

handling these raw packets and dealing with structs and void pointers should happen underneath the api functions

**Comment by SrsBusiness** [comment]
Right now our handlers like this

```c
void position_handler(bot_t *bot, void *autism) {
    my_struct *p = (my_struct *)autism;
    ....
}
```

what we need is this:
```c
void position_handler(bot_t *bot, int x, int y, int z) {
     ....
}
```

handling these raw packets and dealing with structs and void pointers should happen underneath the api functions

**Comment by SrsBusiness** [comment]
Unfortunately this would blow up client.c

**Comment by JDong820** [comment]
Upvoted. Code blow up is inevitable IMO. Welcome to game programming.

**Comment by JDong820** [comment]
Addressed by #91.
